audit_engine/dynamic_analysis/config.py

```
@@ -0,0 +1,161 @@
+            raise ValueError("At least one analysis tool must be enabled")
+        
+        # Validate tool-specific configurations
+        if self.enable_echidna and not isinstance(self.echidna, dict):
+            raise ValueError("Echidna configuration must be a dictionary")
+        
+        if self.enable_adversarial_fuzz and not isinstance(self.adversarial_fuzz, dict):
+            raise ValueError("Adversarial fuzz configuration must be a dictionary")
+            
+        return self
+    
+    @lru_cache(maxsize=1)
```

Ruff (0.12.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)